### PR TITLE
Add support for Mongoid 9

### PR DESCRIPTION
**Pull request by asgerb**
Adds support for Mongoid 9

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ### Unreleased - TBD
 
+* Add support for Mongoid 9 - [@asgerb](https://github.com/asgerb)
```
Comment (dblock) [inline CHANGELOG.md:3]:
```suggestion
* Add support for Mongoid 9 - [@asgerb](https://github.com/asgerb).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ### Unreleased - TBD
 
+* Add support for Mongoid 9 - [@asgerb](https://github.com/asgerb).
```
Comment (dblock) [inline CHANGELOG.md:3]: Sorry to be annoying, Danger wants a link to the PR.

```suggestion
* [#83](https://github.com/mongoid/mongoid-grid_fs/pull/83): Add support for Mongoid 9 - [@asgerb](https://github.com/asgerb).
```

**Review by dblock — CHANGES_REQUESTED** [review]
Looks like MongoDB 7.3 is not a thing on GHA.

Let's use a variety of versions that are supported, maybe list 4, 5, 6, 7, 8 kind of thing across the matrix.

**Comment by asgerb** [comment]
@dblock I've started just with the matrix entries added in this PR, if needed I can make another PR for the rest of the matrix

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -15,6 +15,15 @@ jobs:
...
           - { ruby: '3.2', mongodb: '4.4', gemfile: 'mongoid-8.0' }
+          - { ruby: '3.1', mongodb: '4', gemfile: 'mongoid-9.0' }
+          - { ruby: '3.1', mongodb: '5', gemfile: 'mongoid-9.0' }
+          - { ruby: '3.1', mongodb: '6', gemfile: 'mongoid-9.0' }
+          - { ruby: '3.2', mongodb: '4', gemfile: 'mongoid-9.0' }
+          - { ruby: '3.2', mongodb: '5', gemfile: 'mongoid-9.0' }
+          - { ruby: '3.2', mongodb: '6', gemfile: 'mongoid-9.0' }
+          - { ruby: '3.3', mongodb: '4', gemfile: 'mongoid-9.0' }
+          - { ruby: '3.3', mongodb: '5', gemfile: 'mongoid-9.0' }
+          - { ruby: '3.3', mongodb: '6', gemfile: 'mongoid-9.0' }
```
Comment (dblock) [inline .github/workflows/ci.yml:26]: This is fine, but we don't really need to test every combination of Ruby and MongoDB and mongoid. I would only keep latest 2.x and 3.x, but it's NBD.

**Comment by dblock** [comment]
I released 2.6.0.

If you want to do some more cleanup, upgrades like RuboCop, etc., that would be much welcome.

**Pull request merged**